### PR TITLE
formula-cookbook: document conflicts

### PR DESCRIPTION
**Pull request by DomT4**
Proposal for closing #36179.

### CODE REVIEW

File: share/doc/homebrew/Formula-Cookbook.md [hunk 1]
```diff
@@ -198,6 +198,26 @@ A Hash specifies a formula dependency with some additional information. Given a
...
+
+PolarSSL ship GNU’s Hello, and compiles a `hello` binary. This is obviously non-essential to PolarSSL’s functionality, and conflict with the `hello` formula would be overkill, so we just remove it.
+
+However, also in the PolarSSL formulae is a [firm conflict](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/polarssl.rb#L19) with `md5sha1sum`, because both `md5sha1sum` and `polarssl` compile identically-named binaries that *are* important for core functionality.
+
+As a general rule, `conflicts_with` should be a last-resort option. It’s a fairly blunt instrument.
+
+The syntax for conflict that can’t be worked around is
+
+    ```ruby
```
Comment (DomT4) [inline share/doc/homebrew/Formula-Cookbook.md]: I don't know why the codeblock here isn't Rubyfying. Grr.

Comment (DomT4) [inline share/doc/homebrew/Formula-Cookbook.md]: @mikemcquaid Any idea why the code is Rubyfying under the "Files Changed" tab but when actually [viewed](https://github.com/DomT4/homebrew/blob/docs/share/doc/homebrew/Formula-Cookbook.md) the Ruby colouring vanishes? :confused:

Comment (mikemcquaid) [inline share/doc/homebrew/Formula-Cookbook.md]: Shouldn't be indented.

Comment (DomT4) [inline share/doc/homebrew/Formula-Cookbook.md]: Yup, that fixed it. Thanks!

Weirdly, my local MD editor showed all the Ruby code blocks as indented - I presume my MD editor is just out of sorts.

**Pull request closed without merging**